Fabric.Authorization.AccessControl/src/app/services/global/services.service.ts

```diff
@@ -85,7 +85,25 @@ export class ServicesService {
 
     public needsAuthToken(url: string) {
         const urlLowerCase = url.toLowerCase();
-        const targetService: IService = this.services.find(s => urlLowerCase.includes(s.name.toLowerCase()) && s.requireAuthToken) ? this.services.find(s => urlLowerCase.includes(s.name.toLowerCase()) && s.requireAuthToken) : this.services.find(s => urlLowerCase.startsWith(s.url.toLowerCase()) && s.requireAuthToken);
+        const targetService: IService = this.services.find(s => urlLowerCase.includes(s.name.toLowerCase())) ? this.services.find(s => urlLowerCase.includes(s.name.toLowerCase())) : this.findUrlBestMatch(this.services, url);
         return targetService ? targetService.requireAuthToken : false;
     }
+
+    public findUrlBestMatch(services: IService[], url: string)
+    {
+       var serviceUrlMatch = "";
+       services.forEach(element => {
+        var result = url.toLowerCase().includes(element.url.toLowerCase());
+        if (result)
+        {
+          if (element.url.length > serviceUrlMatch.length)
```

Comment:
I am looping through the 4 services, and taking any matches of the passed in url and setting serviceUrlMatch variable. This isn't always true, since it is possible for it to match on accesscontrol url which is shorter than authorization because of the /v1.